### PR TITLE
[ML][Pipelines] Support component early available output

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Support component early available output, which should be used combining with `is_control`.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/component/input_output.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ class PrimitiveOutputSchema(OutputPortSchema):
         required=True,
     )
     is_control = fields.Bool()
+    early_available = fields.Bool()
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/component/input_output.py:60]: may add a private preview env protection here

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/component/input_output.py]: hide it under `is_private_preview_enabled`
there is one thing different from previous init/finalize: output schema will throw error when private preview env is not set, and as schema classes are always imported at the very beginning (before test class), we need to reload related classes (including schema and entity) in fixture. I did it for `early_available` in new added fixture `enable_early_available_output`.

File: sdk/ml/azure-ai-ml/tests/component/unittests/test_command_component_entity.py [hunk 2]
```diff
@@ -22,6 +23,19 @@
 from .._util import _COMPONENT_TIMEOUT_SECOND
 
 
+@pytest.fixture()
+def enable_early_available_output() -> None:
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/component/unittests/test_command_component_entity.py:27]: may change a different name, this fixture actually does the schema reload not only for early available

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/component/input_output.py [hunk 3]
```diff
@@ -57,7 +58,9 @@ class PrimitiveOutputSchema(OutputPortSchema):
         required=True,
     )
     is_control = fields.Bool()
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/component/input_output.py:60]: may also hide is_control?

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/component/input_output.py:60]: sure; this change may break some tests for schema issue, handling them may take some time (add fixture here and there)

File: sdk/ml/azure-ai-ml/tests/component/unittests/test_command_component_entity.py (hunk 2 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/component/unittests/test_command_component_entity.py]: updated to `enable_private_preview_schema_features`, same as Azure DevOps comment

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_inputs_outputs/output.py [hunk 4]
```diff
@@ -80,14 +80,16 @@ def __init__(self, *, type=AssetTypes.URI_FOLDER, path=None, mode=None, descript
         self.mode = mode
         # use this field to determine the Output is control or not, currently hide in kwargs
         self.is_control = kwargs.pop("is_control", None)
+        # use this field to mark Output for early node orchestrate, currently hide in kwargs
+        self.early_available = kwargs.pop("early_available", None)
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_inputs_outputs/output.py:84]: Is there a doc explaining what do `is_control` and `early_available` mean?

Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_inputs_outputs/output.py:84]: both of these fields are for pipelines only internal feature; to be more specific, `is_control` is for if-else condition and some other scenarios, `early_available` is for some upcoming features in pipelines.
if you are interested in the related pipelines feature, please ping me or @wangchao1230 , maybe some internal doc can kind of solve your confusion.

**Pull request merged**